File: ios/MenuMakerUITests/PageObjects/ForgotPasswordPage.swift
//
//  ForgotPasswordPage.swift
//  MenuMakerUITests
//
//  Page Object for Forgot Password Screen
//

import XCTest

struct ForgotPasswordPage {
    let app: XCUIApplication

    // MARK: - Elements

    private var forgotPasswordScreen: XCUIElement {
        app.scrollViews["forgot-password-screen"]
    }

    var emailField: XCUIElement {
        forgotPasswordScreen.textFields["email-field"]
    }

    var submitButton: XCUIElement {
        app.buttons["submit-button"]
    }

    var backButton: XCUIElement {
        app.navigationBars.buttons.firstMatch
    }

    var successMessage: XCUIElement {
        app.staticTexts["success-message"]
    }

    var errorMessage: XCUIElement {
        app.staticTexts["error-message"]
    }

    // MARK: - Actions

    @discardableResult
    func enterEmail(_ email: String) -> ForgotPasswordPage {
        // Wait for field to be hittable before interacting
        XCTAssertTrue(emailField.waitForExistence(timeout: 3), "Email field should exist")

        emailField.tap()
        emailField.typeText(email)
        return self
    }

    @discardableResult
    func tapSubmit() -> ForgotPasswordPage {
        // No need to dismiss keyboard - tapping button will dismiss it automatically
        submitButton.tap()
        return self
    }

    @discardableResult
    func tapBack() -> LoginPage {
        backButton.tap()
        return LoginPage(app: app)
    }

    // MARK: - Assertions

    @discardableResult
    func assertScreenDisplayed(timeout: TimeInterval = 2) -> ForgotPasswordPage {
        XCTAssertTrue(forgotPasswordScreen.waitForExistence(timeout: timeout), "Forgot password screen should be displayed")
        return self
    }

    @discardableResult
    func assertSuccessDisplayed(timeout: TimeInterval = 3) -> ForgotPasswordPage {
        XCTAssertTrue(successMessage.waitForExistence(timeout: timeout), "Success message should be displayed")
        return self
    }

    @discardableResult
    func assertErrorDisplayed(timeout: TimeInterval = 3) -> ForgotPasswordPage {
        XCTAssertTrue(errorMessage.waitForExistence(timeout: timeout), "Error message should be displayed")
        return self
    }

    // MARK: - Helpers

    private func dismissKeyboardIfNeeded() {
        if app.keyboards.count > 0 {
            app.keyboards.buttons["Return"].tap()
        }
    }
}

File: android/app/src/androidTest/kotlin/com/menumaker/pageobjects/ForgotPasswordPage.kt
package com.menumaker.pageobjects

import androidx.compose.ui.test.*
import androidx.compose.ui.test.junit4.ComposeTestRule

/**
 * Page Object for Forgot Password Screen
 * Provides fluent API for password reset interactions
 */
class ForgotPasswordPage(private val composeTestRule: ComposeTestRule) {

    // Elements
    private val emailField = composeTestRule.onNodeWithTag("email-field")
    private val submitButton = composeTestRule.onNodeWithTag("submit-button")
    private val backButton = composeTestRule.onNodeWithContentDescription("Back", substring = true)
    private val successMessage = composeTestRule.onNodeWithTag("success-message")
    private val errorMessage = composeTestRule.onNodeWithTag("error-message")

    // Actions
    fun enterEmail(email: String): ForgotPasswordPage {
        emailField.performTextInput(email)
        return this
    }

    fun tapSubmit(): ForgotPasswordPage {
        submitButton.performClick()
        Thread.sleep(2000) // Wait for API response
        return this
    }

    fun tapBack(): LoginPage {
        backButton.performClick()
        Thread.sleep(500)
        return LoginPage(composeTestRule)
    }

    // Assertions
    fun assertScreenDisplayed(): ForgotPasswordPage {
        emailField.assertExists()
        submitButton.assertExists()
        return this
    }

    fun assertSuccessDisplayed(): ForgotPasswordPage {
        successMessage.assertExists()
        return this
    }

    fun assertErrorDisplayed(): ForgotPasswordPage {
        errorMessage.assertExists()
        return this
    }

    fun assertSubmitButtonEnabled(): ForgotPasswordPage {
        submitButton.assertIsEnabled()
        return this
    }

    fun assertSubmitButtonDisabled(): ForgotPasswordPage {
        submitButton.assertIsNotEnabled()
        return this
    }
}
